CFile.js

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+
+var logger = log4js.getLogger("loctool.plugin.CFile");
+
+/**
+ * Create a new java file with the given path name and within
+ * the given project.
+ *
+ * @param {Project} project the project object
+ * @param {String} pathName path to the file relative to the root
+ * of the project file
+ * @param {FileType} type the file type of this instance
+ */
+var CFile = function(props) {
+    this.project = props.project;
+    this.pathName = props.pathName;
+    this.type = props.type;
+    this.API = props.project.getAPI();
+
+    this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+};
+
+/**
+ * Unescape the string to make the same string that would be
+ * in memory in the target programming language.
+ *
+ * @static
+ * @param {String} string the string to unescape
+ * @returns {String} the unescaped string
+ */
+CFile.unescapeString = function(string) {
+    var unescaped = string;
+
+    unescaped = unescaped.
+        replace(/\\\\n/g, "").                // line continuation
+        replace(/\\\n/g, "").                // line continuation
+        replace(/^\\\\/, "\\").             // unescape backslashes
+        replace(/([^\\])\\\\/g, "$1\\").
+        replace(/^\\'/, "'").               // unescape quotes
+        replace(/([^\\])\\'/g, "$1'").
+        replace(/^\\"/, '"').
+        replace(/([^\\])\\"/g, '$1"');
+
+    return unescaped;
+};
+
+/**
+ * Clean the string to make a resource name string. This means
+ * removing leading and trailing white space, compressing
+ * whitespaces, and unescaping characters. This changes
+ * the string from what it looks like in the source
+ * code but increases matching.
+ *
+ * @static
+ * @param {String} string the string to clean
+ * @returns {String} the cleaned string
+ */
+CFile.cleanString = function(string) {
+    var unescaped = CFile.unescapeString(string);
+
+    unescaped = unescaped.
+        replace(/\\[btnfr]/g, " ").
+        replace(/[ \n\t\r\f]+/g, " ").
+        trim();
+
+    return unescaped;
+};
+
+/**
+ * Make a new key for the given string. This must correspond
+ * exactly with the code in htglob jar file so that the
+ * resources match up. See the class IResourceBundle in
+ * this project under the java directory for the corresponding
+ * code.
+ *
+ * @private
+ * @param {String} source the source string to make a resource
+ * key for
+ * @returns {String} a unique key for this string
+ */
+CFile.prototype.makeKey = function(source) {
+    return CFile.unescapeString(source);
+};
+
+var reGetLocString = new RegExp(/\bresBundle_getLocString\((.*)\,\s*[\"|\'](.*)*[\"|\']\)\;/g);
+var reGetLocStringWithKey = new RegExp(/\bresBundle_getLocStringWithKey\((.*)\,\s*[\"|\'](.*)[\"|\']\,\s*[\"|\'](.*)*[\"|\']\)/g);
+var reI18nComment = new RegExp("//\\s*i18n\\s*\\s*(.*)$");
```

Comment:
C does not support double-slash comments. It only supports /* this style */ of comments.

Reply:
Yeah.. but we have cases to use double-slash in i18n comments. I think I need to support follow three styles.
```
1. // i18n comments hello
2. /* i18n comments hello */
3. /* i18n comments
    *  hello
    */
```

Reply:
The double-slash comments should not appear in valid C files. It's probably okay to leave the support for it, but it is not needed.

Reply:
I've been thinking to support multi-line i18n comments, It's a bit weird. I think only a single-line comment would be fine. What do you think?
and I checked some C service codes in webOS. most of the comments were written double-slash style. That comment form doesn't occur any logic problem, and there wasn't a specific guideline so far, so I assume they just following JS i18n comment style..
They were some examples that found:
```
// i18n  /** Connect WiSA Dongle **/ 
// i18n : Power button
// i18n : GUIDE button for Set-top box, used in screen remote
```